pkg/gitops/profile.go

```diff
@@ -0,0 +1,138 @@
+package gitops
+
+import (
+	"context"
+	"github.com/kris-nova/logger"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+	"github.com/weaveworks/eksctl/pkg/git"
+	"github.com/weaveworks/eksctl/pkg/gitops/fileprocessor"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	cloneDirPrefix = "quickstart-"
+)
+
+// Profile represents a GitOps profile
+type Profile struct {
+	Processor fileprocessor.FileProcessor
+	Path      string
+	GitOpts   GitOptions
+	GitCloner git.Cloner
+	Fs        afero.Fs
+	IO        afero.Afero
```

Comment:
Why not call it `Aero`?

Reply:
I just called it IO because that's what we had in the previous version of this PR. I can rename it.

Reply:
Actually I thought about it makes more sense that it has a more generic name related to what it does rather than the library it uses. so unless there are strong opinions I would like to keep it as `IO`.